src/Microsoft.FeatureManagement/FeatureManagerSnapshot.cs

```diff
@@ -24,53 +24,78 @@ public FeatureManagerSnapshot(IFeatureManager featureManager)
         public async IAsyncEnumerable<string> GetFeatureNamesAsync()
         {
             if (_featureNames == null)
+            {
+                _featureNames = await CreateListAsync(_featureManager).ConfigureAwait(false);
```

Comment:
Didn't fix this race, as
* I assume this method isn't hot
* in the case of a race the list is created multiple time and `_featureNames` will get overwritten (in atomic way due being a reference), so unnecessary lists are just thrown away and are eligible for GC

Reply:
Why did this method get touched?

Reply:
I can't remember the exact motivation for the change, but in the case `_featureNames != null` the method will be way shorter, as when the lazy-init logic is part of the default-codepath. Aka hot-/cold-path optimization.

Back then when I made the change, I just did some optimization, so maybe I was a bit too aggressive in optimizing. Now I don't know if this change is worth it.

Revert this one?

Reply:
Yes, I would revert this one.

Reply:
Reverted with b4855a9ea2f9982f839ec9c06655e1e89c061b2f